### PR TITLE
[onewire] Add support for BAE091x

### LINKED ISSUE #4916: [onewire] support BAE0910 sensors
add support for BAE0910 sensors in the onewrie binding

was ESH [#6177](https://github.com/eclipse/smarthome/issues/6177)

### PR DESCRIPTION
**Pull request by J-N-K**
Resolves #4916 
Depends on #4710

### CODE REVIEW

**Comment by wborn** [comment]
Is there still anything to be done except for fixing the merge conflict now #4710 is merged?

**Comment by J-N-K** [comment]
This is totally untested. I’ll relive all conflicts now and seeing I can get the requester to test it.

 Since it is mostly new files, I can easily fix if this is breaking after changing the build system.

**Comment by davidgraeff** [comment]
Can you please resolve the conflicts?

Because no one can perform functional tests for this onewire extension, and it also doesn't break any existing functionality I'd say we merge as soon as travis agrees.

File: addons/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/handler/BAE091xSensorThingHandler.java [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+                break;
+            case CONFIG_BAE_PIN_PWM:
+                wantedChannel
+                        .add(new OwChannelConfig(CHANNEL_PWM_DUTY2, CHANNEL_TYPE_UID_BAE_PWM_DUTY, "Duty Cycle PWM 2"));
+                wantedChannel.add(new OwChannelConfig(CHANNEL_PWM_FREQ2, CHANNEL_TYPE_UID_BAE_PWM_FREQUENCY,
+                        "Frequency PWM 2/4"));
+                break;
+            default:
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "unknown configuration option for pin 2");
```
Comment (davidgraeff) [inline addons/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/handler/BAE091xSensorThingHandler.java]: pin 2 although in pin 7 switch case?

File: addons/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/handler/BAE091xSensorThingHandler.java [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+                wantedChannel.add(new OwChannelConfig(CHANNEL_COUNTER, CHANNEL_TYPE_UID_BAE_COUNTER));
+                break;
+            default:
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "unknown configuration option for pin 1");
+                return;
+        }
+
+        // Pin2:
+        switch (configuration.pin2) {
```
Comment (davidgraeff) [inline addons/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/handler/BAE091xSensorThingHandler.java]: Is it possible to express this switch case in a simpler way? A lot of cases look very similar and I have the impression that a static configuration lookup table for those pins makes more sense?

File: addons/binding/org.openhab.binding.onewire/ESH-INF/thing/bae09x.xml [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+                <options>
+                    <option value="input">Input</option>
+                    <option value="output">Output</option>
+                </options>
+                <default>input</default>
+                <limitToOptions>true</limitToOptions>
+            </parameter>
+            <parameter name="pulldevice" type="text">
+                <label>Pull-Up/Pull-Down Resistor</label>
+				<options>
```
Comment (davidgraeff) [inline addons/binding/org.openhab.binding.onewire/ESH-INF/thing/bae09x.xml]: formatting

**Comment by davidgraeff** [comment]
Still WIP?

**Comment by J-N-K** [comment]
Yes, missing tests and documentation. But i‘ll try to get that finished ASAP.

**Comment by J-N-K** [comment]
Finished. Also fixed some bugs found with the tests.

**Pull request merged**